packages/language-support/src/highlighting/syntaxValidation/semanticAnalysisWrapper.ts

```diff
@@ -29,7 +31,27 @@ export function doSemanticAnalysis(query: string): SemanticAnalysisResult {
     const errors: SemanticAnalysisElement[] = semanticErrorsResult.$errors.data;
     const notifications: SemanticAnalysisElement[] =
       semanticErrorsResult.$notifications.data;
-    return { errors: errors, notifications: notifications };
+
+    return {
+      errors: errors.map(({ message, position }) => ({
+        severity: DiagnosticSeverity.Error,
+        message,
+        position: {
+          column: position.column,
+          line: position.line,
```

Comment:
This gets rid of some $$position objects that could not be passed over the `webworker` <-> `main thread` boundary. I also add the severity here, so we can make the mapping later simpler